bundles/automation/org.eclipse.smarthome.automation.core/src/main/java/org/eclipse/smarthome/automation/core/RuleProvider.java

```diff
@@ -0,0 +1,59 @@
+package org.eclipse.smarthome.automation.core;
+
+import java.util.Collection;
+
+import org.eclipse.smarthome.automation.Rule;
+import org.eclipse.smarthome.core.common.registry.AbstractProvider;
+import org.eclipse.smarthome.core.common.registry.ManagedProvider;
+import org.osgi.framework.BundleContext;
+
+/**
+ *
+ * @author Yordan Mihaylov - Initial Contribution
+ */
+public class RuleProvider extends AbstractProvider implements ManagedProvider {
```

Comment:
this is what we tried initially but we had issues with the storage service not working on Qivicon (maybe it was a setup issue) But since we couldn't see it working we postponed the storage feature for the next delivery. Maybe you can help us with the setup of the runtime ?